Fix access mode e2e tests and add more test cases

DaoDaoNoCode opened the pull request:
<!--- If this is a non-code change, this template is not required; reference any issues or top-level descriptions as needed -->
<!--- All code change PRs should relate to an issue, reference it here; see example below -->
<!--- https://issues.redhat.com/browse/RHOAIENG-123456 -->
JIRA: https://issues.redhat.com/browse/RHOAIENG-33410

## Description
<!--- Describe your changes in detail; the what, the why, any findings, etc -->
<!--- Include any screenshots of changed UI; Include any gifs if it was a flow / UX change -->
Fix the e2e tests for the access mode, and add some test cases to both cluster storage table and the workbench spawner page.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Run e2e test.

## Test Impact
<!--- What tests have you done to cover the implemented functionality -->
<!--- If tests are not applicable, explain why here -->
N/A

## Request review criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)
- [x] The code follows our [Best Practices](/docs/best-practices.md) (React coding standards, PatternFly usage, performance considerations)

If you have UI changes: 
<!--- You can ignore these if you are doing manifest, backend, internal logic, etc changes; aka non-UI / visual changes -->
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Tests**
  * Enhanced e2e coverage for storage access-mode behavior across cluster storage and workbench workflows
  * Converted tests to use data-driven fixtures with typed test data and added helpers to drive selection, creation, attachment and verification of access modes
  * Added scenarios validating access-mode display and resetting when switching storage classes
* **Chores**
  * Added YAML test fixtures and updated test lint ignore rules; improved fixture loading utilities
* **Bug Fixes**
  * Made storage-class deletion idempotent during cleanup

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

DaoDaoNoCode commented:
/test-e2e

antowaddle commented on frontend/src/__tests__/cypress/cypress/tests/e2e/dataScienceProjects/workbenches/testWorkbenchStorageClasses.cy.ts:
```diff
@@ -114,4 +116,200 @@ describe('Workbench Storage Classes Tests', () => {
+      provisionStorageClass(scMultiAccessName, StorageProvisioner.AZURE_FILE, scMultiAccess);
+      createdStorageClasses.push(scMultiAccessName);
+
+      const rwoMountPath = 'rwo-mount-path';
```
Same with these, please add to fixtures.

antowaddle commented on frontend/src/__tests__/cypress/cypress/tests/e2e/dataScienceProjects/workbenches/testWorkbenchStorageClasses.cy.ts:
```diff
@@ -114,4 +116,200 @@ describe('Workbench Storage Classes Tests', () => {
+          });
+
+          // Verify ROX storage exists with ReadOnlyMany
+          cy.contains('tr', storageROX).within(() => {
```
Can we add this to a page object?

DaoDaoNoCode commented:
@antowaddle Hi, I updated the tests according to your requests. I also updated the `.eslintignore` file in case it throws lint error on those invalid files. Let me know if you have any questions, thank you!
<img width="1387" height="2770" alt="Screenshot 2025-11-20 at 5 59 23 PM" src="https://github.com/user-attachments/assets/7dabd885-9536-4925-b72e-f40fef287991" />
<img width="1404" height="3322" alt="Screenshot 2025-11-20 at 6 58 43 PM" src="https://github.com/user-attachments/assets/06b2e834-caf2-4fe7-8ef8-2d99b14e925c" />

antowaddle commented on frontend/src/__tests__/cypress/cypress/tests/e2e/dataScienceProjects/workbenches/testWorkbenchStorageClasses.cy.ts:
```diff
@@ -45,35 +56,35 @@ describe('Workbench Storage Classes Tests', () => {
 
   it(
     'Create workbench with RWO storage and verify storage attachment',
     { tags: ['@Storage', '@ODS-1931', '@Dashboard', '@Workbenches'] },
```
Can you please add smoke/sanity tags for these tests? Please review other related tests in the repository to catch a glimpse of the types of tagging in use.

antowaddle commented on frontend/src/__tests__/cypress/cypress/tests/e2e/dataScienceProjects/workbenches/testWorkbenchStorageClasses.cy.ts:
```diff
@@ -45,35 +56,35 @@ describe('Workbench Storage Classes Tests', () => {
-      const workbenchName = 'wb-rwo-storage';
-      const storageName = 'rwo-storage';
-      const mountPath = 'mnt/different-path';
+      const workbenchName = testData.workbenchRWO;
```
If you already import the fixture, why not just declare these variables then and access them throughout the test? Can you review other e2e tests to view how we do that as a general pattern?

antowaddle commented on frontend/src/__tests__/cypress/cypress/utils/clusterStorage.ts:
```diff
@@ -3,14 +3,29 @@ import { clusterStorage } from '#~/__tests__/cypress/cypress/pages/clusterStorag
-    if ($body.find('[data-testid="cluster-storage-button"]').length > 0) {
+export const findAddClusterStorageButton = (): Cypress.Chainable<JQuery<HTMLElement>> => {
+  // Check which button exists in the DOM by querying both page object methods
+  return cy.get('body').then(() => {
```
Can we look at using testIDs for the locators here please?
